airbyte-cdk/python/airbyte_cdk/connector_builder/message_grouper.py

```diff
@@ -176,12 +180,32 @@ def _need_to_close_page(at_least_one_page_in_group: bool, message: AirbyteMessag
         return (
             at_least_one_page_in_group
             and message.type == MessageType.LOG
-            and (MessageGrouper._is_page_request_response_log(json_message) or message.log.message.startswith("slice:"))
+            and (MessageGrouper._is_page_http_request(json_message) or message.log.message.startswith("slice:"))
         )
 
     @staticmethod
-    def _is_page_request_response_log(message: Optional[dict]):
-        return message and message.get("http") and message.get("log", {}).get("logger", "") == SimpleRetriever.LOGGER_NAME
+    def _is_page_http_request(json_message):
+        return MessageGrouper._is_http_log(json_message) and not MessageGrouper._is_global_request(json_message)
+
+    @staticmethod
+    def _is_http_log(message: Optional[dict]) -> bool:
+        return message and bool(message.get("http", False))
+
+    @staticmethod
+    def _is_global_request(message: Optional[dict]) -> bool:
+        """
+        A global request is a request that is performed and will not directly lead to record for the specific stream it is being queried. A
+        couple of examples are:
+        * OAuth authentication
+        * Substream slice generation
+        """
+        if not message:
+            return False
+
+        is_http = MessageGrouper._is_http_log(message)
+        has_stream_name = message.get("airbyte_cdk", {}).get("stream", {}).get("name", False)
+        is_substream = message.get("airbyte_cdk", {}).get("stream", {}).get("is_substream", False)
```

Comment:
There are currently two cases:
* Global requests not associated to a stream: Authentication is one example of this i.e. that the authenticator doesn't exactly know for which stream it does the request. We could even re-use the same authenticator across streams (which I think today we don't do today but it's really not far-fetched) 
* Global requests for substream: in that case, the `is_substream` flag will be set to true

So to answer the question "Will it always be the case that these types of requests won't have a stream name associated with them?", it's not always the case today as substream requests are considered "global" but have a stream name. This is why the condition is `is_http and (not has_stream_name or is_substream)` If you think it can be more obvious from the HTTP log that a request is return record for the stream or not, let me know. I know we discussed that we might assume from the message grouper that test reads are always for one stream only and if the message grouper knows which stream, we can filter based on that but we could avoid assuming this for now so I didn't used this solution

Reply:
I was thinking more that we should know based on the context whether the request is for a stream-specific operation or not. The fact that a stream name is present or not feels somewhat tangential to whether the request is a "supporting request" versus actually getting data, so I was hoping that there was a more direct way to identify these requests. However, this is not a blocking comment.

Reply:
Would a field `http.is_global` or `http.is_auxiliary` makes sense to you? It would mean that the `Oauth2Authenticator` would call `format_http_message` with `auxiliary=True` and for the substream, we would have the log appender overwriting this `http.is_auxiliary` value

Reply:
Yes that would make sense to me.

Reply:
Done! Thank you for this feedback. I also think it makes things much clearer